addons/mail/models/res_users_settings_volumes.py

```diff
@@ -14,6 +14,26 @@ class ResUsersSettingsVolumes(models.Model):
     guest_id = fields.Many2one('res.partner', ondelete='cascade', index=True)
     volume = fields.Float(default=0.5, help="Ranges between 0.0 and 1.0, scale depends on the browser implementation")
 
+    def _mail_users_settings_volume_format(self):
+        records = []
+        for record in self:
+            vals = {
+                'id': record.id,
+                'volume': record.volume,
+            }
+            if record.guest_id:
+                vals['guest'] = [('insert', {
+                    'id': record.guest_id.id,
+                    'name': record.guest_id.name,
+                })]
+            else:
+                vals['partner'] = [('insert', {
```

Comment:
Also it would be better in general if instead of not having the key for falsy fields, we did say explicitly that the field is falsy (so instead of nothing, it would be clear command when it's not set).
It doesn't matter much in this case as the field value is not supposed to change, but it's good to have the habit.